src/main/java/com/minecolonies/coremod/colony/buildings/AbstractFilterableListCrafter.java

```diff
@@ -0,0 +1,214 @@
+            Collectors.toList());
+        final ImmutableList.Builder<IRequestResolver<?>> builder = ImmutableList.builder();
+
+        builder.addAll(supers);
+        builder.add(new PublicWorkerCraftingRequestResolver(getRequester().getLocation(),
+          getColony().getRequestManager().getFactoryController().getNewInstance(TypeConstants.ITOKEN)));
+        builder.add(new PublicWorkerCraftingProductionResolver(getRequester().getLocation(),
+          getColony().getRequestManager().getFactoryController().getNewInstance(TypeConstants.ITOKEN)));
+
+        return builder.build();
+    }
+
+    @Override
+    public Map<Predicate<ItemStack>, Tuple<Integer, Boolean>> getRequiredItemsAndAmount()
+    {
+        final Map<ItemStorage, Tuple<Integer, Boolean>> recipeOutputs = new HashMap<>();
+        for (final ICitizenData citizen : getAssignedCitizen())
+        {
+            if (citizen.getJob() instanceof AbstractJobCrafter)
+            {
+                final List<IToken<?>> assignedTasks = citizen.getJob(AbstractJobCrafter.class).getAssignedTasks();
+                for (final IToken taskToken : assignedTasks)
+                {
+                    final IRequest<? extends PublicCrafting> request = (IRequest<? extends PublicCrafting>) colony.getRequestManager().getRequestForToken(taskToken);
+                    final IRecipeStorage recipeStorage = getFirstRecipe(request.getRequest().getStack());
+                    if (recipeStorage != null)
+                    {
+                        for (final ItemStorage itemStorage : recipeStorage.getCleanedInput())
+                        {
+                            int amount = itemStorage.getAmount();
+                            if (recipeOutputs.containsKey(itemStorage))
+                            {
+                                amount = recipeOutputs.get(itemStorage).getA() + itemStorage.getAmount();
+                            }
+                            recipeOutputs.put(itemStorage, new Tuple<>(amount, false));
+                        }
+
+                        final ItemStorage output = new ItemStorage(recipeStorage.getPrimaryOutput());
+                        if (recipeOutputs.containsKey(output))
+                        {
+                            output.setAmount(recipeOutputs.get(output).getA() + output.getAmount());
+                        }
+                        recipeOutputs.put(output, new Tuple<>(output.getAmount(), false));
+                    }
+                }
+            }
+        }
+
+        final Map<Predicate<ItemStack>, Tuple<Integer, Boolean>> toKeep = new HashMap<>(keepX);
+        toKeep.putAll(recipeOutputs.entrySet().stream().collect(Collectors.toMap(key -> (stack -> stack.isItemEqual(key.getKey().getItemStack())), Map.Entry::getValue)));
+        return toKeep;
+    }
+
+    @Override
+    public boolean canCraftComplexRecipes()
```

Comment:
I guess we don't need those methods here (canCraft, canRecipeBeAdded, openCraftingcontainer)

Reply:
The `AbstractFilterableListCrafter` inherits from `AbstractFilterableListBuilding`, not from `AbstractBuildingCrafter`. (The FilterableList one is tighter coupled with messages, so it seemed less error-prone to me to do it this way). Removing the aforementioned methods would lock this type out from complex recipes, which is not necessarily what we want. I'd leave it in for now - this whole thing (Abstract class for every type + their combinations) needs to be reworked in the long run anyway imho.

Reply:
Ah okay, while it might not be necessary for the current lj, others might need it, true.

Reply:
I did realize that `openCraftingContainer` was already useless ('cause 1:1 copy of its parent) in the original `AbstractBuildingCrafter`. I removed it from both there and from the new `AbstractFilterableListCrafter`. Fixed in f1bab0e.